src/main/java/plugily/projects/villagedefense/events/Events.java

```diff
@@ -433,39 +436,53 @@ public void onCraft(CBPlayerInteractEvent event) {
   }
 
   @EventHandler
-  public void onRottenFleshDrop(InventoryPickupItemEvent e) {
+  public void onSecretWellDrop(InventoryPickupItemEvent e) {
     if(e.getInventory().getType() != InventoryType.HOPPER) {
       return;
     }
-    if(e.getItem().getItemStack().getType() != Material.ROTTEN_FLESH) {
-      for(Arena arena : ArenaRegistry.getArenas()) {
-        if(e.getItem().getWorld().equals(arena.getStartLocation().getWorld())) {
-          e.getItem().remove();
-          e.getInventory().clear();
-          return;
-        }
+
+    Item item = e.getItem();
+    ItemStack itemStack = item.getItemStack();
+    Location location = item.getLocation();
+
+    Arena currentArena = null;
+    for(Arena arena : ArenaRegistry.getArenas()) {
+      if(item.getWorld().equals(arena.getStartLocation().getWorld())) {
+        currentArena = arena;
+        item.remove();
+        e.setCancelled(true);
+        e.getInventory().clear();
+        break;
       }
+    }
+    if (currentArena == null) {
       return;
     }
-    for(Entity entity : Utils.getNearbyEntities(e.getItem().getLocation(), 20)) {
-      if(!(entity instanceof Player)) {
-        continue;
-      }
-      Arena arena = ArenaRegistry.getArena((Player) entity);
-      if(arena == null) {
-        continue;
-      }
-      arena.addOptionValue(ArenaOption.ROTTEN_FLESH_AMOUNT, e.getItem().getItemStack().getAmount());
-      e.getItem().remove();
-      e.setCancelled(true);
-      e.getInventory().clear();
```

Comment:
Why isn't that readded on the new part? (Or did you do tests and it worked well without it?)

Reply:
It works without it. We have already cleared it when checking arenas.